src/utils/so/YamlIo.cc

```diff
@@ -9,13 +9,257 @@
+                    out = escape_sequence('e', out);
+                    break;
+                // case 0x0020: // space
+                //    out = escape_sequence('', out);
+                //    break;
+                case 0x0022: // double quote
+                    out = escape_sequence('"', out);
+                    break;
+                // case 0x002F: // slash
+                //    out = escape_sequence('/', out);
+                //    break;
+                case 0x005C: // back slash
+                    out = escape_sequence('\\', out);
+                    break;
+                case 0x0085: // utf next line
+                    out = escape_sequence('N', out);
+                    break;
+                case 0x00A0: // utf non-breaking space
+                    out = escape_sequence('_', out);
+                    break;
+                case 0x2028: // utf line separator
+                    out = escape_sequence('L', out);
+                    break;
+                case 0x2029: // utf paragraph separator
+                    out = escape_sequence('P', out);
+                    break;
+                default: {
+                    if (is_yaml_printable(c)) {
+                        out = serialize(c, out);
+                    } else {
+                        out = serialize_escaped(c, out);
+                    }
+                }
+            }
+        }
+        return out;
+    }
+
+    std::ostream& serialize_yaml(std::ostream& out, const std::string& obj)
+    {
+        out << '"';
+        escape_yaml_string( //
+            obj.begin(),
+            obj.end(),
+            std::ostream_iterator<char>(out));
+        out << '"';
+        return out;
+    }
+
+    std::ostream& do_indent(std::ostream& out, int indent)
+    {
+        for (; indent > 0; --indent)
+            out << "  ";
+        return out;
+    }
+
+    struct yaml_printer final {
+        void operator()(const Null& value, std::ostream& out, int indent = 0) const
+        {
+            if (indent > 0)
+                out << ' ';
+
+            out << "null";
+        }
+
+        void operator()(const True& value, std::ostream& out, int indent = 0) const
+        {
+            if (indent > 0)
+                out << ' ';
+
+            out << "true";
+        }
+
+        void operator()(const False& value, std::ostream& out, int indent = 0) const
+        {
+            if (indent > 0)
+                out << ' ';
+
+            out << "false";
+        }
+
+        void operator()(const String& value, std::ostream& out, int indent = 0) const
+        {
+            if (indent > 0)
+                out << ' ';
+
+            serialize_yaml(out, value.data);
+        }
+
+        void operator()(const Number& value, std::ostream& out, int indent = 0) const
+        {
+            if (indent > 0)
+                out << ' ';
+
+            out << value.data; // TODO @tjanc@ probably not correct
```

Comment:
What does the YAML spec say? JSON needed the scientific stuff IIRC

Reply:
From what I gathered YAML accepts default C++ `double` serialization. Will remove TODO.